src/mesh.cpp

```diff
@@ -2568,7 +2561,24 @@ double LibMesh::volume(int bin) const
 
 void read_meshes(pugi::xml_node root)
 {
+  std::unordered_set<int> mesh_ids;
+
   for (auto node : root.children("mesh")) {
+    // Check to make sure multiple meshes in the same file don't share IDs
+    int id = std::stoi(get_node_value(node, "id"));
+    if (contains(mesh_ids, id)) {
+      fatal_error(
+        "Two or more meshes use the same unique ID: " + std::to_string(id));
```

Comment:
We'll want to clarify here that these meshes are in the same file. Otherwise, the difference between triggering the fatal error here and the warning below may be confusing to users.

```suggestion
        fmt::format("Two or more meshes use the same unique ID '{}' in the same input file", id));
```